### DISCUSSION TITLE
Removing a file from commit

### DISCUSSION THREAD

**Discussion by bennypowers** [jesseduffield/lazygit#2281] [Q&A] [answered]
A workflow I occasionally have to do happens when I realize that a commit somewhere in my log has unrelated changes - those changes will need to be extracted to their own commit. Let's say I have two commits, one which installs a tool to optimize svg files and a second later commit which actually optimizes the svg files in the repo. while inspecting the second commit i notice there are changes to package.json - whoops! I need to move that change to the previous commit, and maybe there are other commits in between, complicating things.

So i'll open lazygit, navigate to the commit in question, and hit `e` to enter edit mode, but at this point I'm still kinda on my own, because the staging area is empty, so I'll open a new command line and 

```
git reset --soft HEAD~1
git reset HEAD package.json
git commit -m "fixup: optimize svgs"
git add package.json
git commit -m "fixup: install svgo"
```

I'd prefer to do this workflow entirely in lazygit. Are there tools to do this already that I just missed out on?

**Comment by jesseduffield** [comment, marked as answer]
You can select the files you wanna bring across by pressing space and then if you go Ctrl+p there's an option to move those files to another commit. That ought to do what you want :)

**Reply by bennypowers** [reply]
i'm missing something, if the staging area is empty, where do i select the files?

**Reply by jesseduffield** [reply]
if you press enter on the commit you can see it's files. From within there you can select the files

**Reply by bennypowers** [reply]
gorgeous!
